mlir/test/Dialect/Linalg/vectorize-tensor-extract.mlir

```diff
@@ -253,6 +253,58 @@ module attributes {transform.with_named_sequence} {
     transform.yield
   }
 }
+
+// -----
+
+#map = affine_map<(d0, d1) -> (d0, d1)>
+#map1 = affine_map<(d0, d1, d2) -> (d0 + d1 + d2)>
+func.func @vectorize_nd_tensor_extract_without_outer_unit_dim(%arg0: tensor<8x128x768xf32>, %arg1 : index) -> tensor<8x1xf32> {
+  %c0 = arith.constant 0 : index
+  %0 = tensor.empty() : tensor<8x1xf32>
+  %1 = linalg.generic {
+    indexing_maps = [#map], 
+    iterator_types = ["parallel", "parallel"]
+  } outs(%0 : tensor<8x1xf32>) {
+  ^bb0(%arg5: f32):
+      %2 = linalg.index 0 : index
+      %3 = linalg.index 1 : index
+      %4 = affine.apply #map1(%arg1, %3, %arg1)
+    %extracted = tensor.extract %arg0[%2, %c0, %4] : tensor<8x128x768xf32>
+    linalg.yield %extracted : f32
+  } -> tensor<8x1xf32>
```

Comment:
Let's try to simplify this example, while making sure similar cases is tested. From what I can tell, the following generic will exercise exactly the same case, right?

```suggestion
  %1 = linalg.generic {
    indexing_maps = [#map], 
    iterator_types = ["parallel", "parallel"]
  } outs(%0 : tensor<8x1xf32>) {
  ^bb0(%arg5: f32):
      %2 = linalg.index 0 : index
    %extracted = tensor.extract %arg0[%2, %c0] : tensor<8x128xf32>
    linalg.yield %extracted : f32
  } -> tensor<8x1xf32>
```

Reply:
Ignore this suggestion, it's hitting a slightly different case. I can take care of this in a follow-on patch.